src/tabs/src/Tabs.tsx

```diff
@@ -389,6 +389,33 @@ export default defineComponent({
       barEl.classList.remove(disableTransitionClassName)
     }
 
+    const tabsRailElRef = ref<HTMLElement | null>(null)
+    const segmentCapsuleElRef = ref<HTMLElement | null>(null)
+
+    function updateSegmentPosition (): void {
+      if (props.type === 'segment') {
+        const { value: tabsRailEl } = tabsRailElRef
+        if (!tabsRailEl) return
+        void nextTick(() => {
+          tabsRailEl.classList.add('transition-disabled')
+          const activeTabEl = tabsRailEl.querySelector('.n-tabs-tab--active')
```

Comment:
The way that select active tab need to be refactored, query selector is not preferred.